### PR TITLE
Lagom 1.6.0-RC3

### PR DESCRIPTION
**Pull request by octonato**

### CODE REVIEW

File: grpc-example/grpc-example-scala/build.sbt [hunk 1]
```diff
@@ -1,4 +1,8 @@
-import akka.grpc.gen.scaladsl.play.{ PlayScalaClientCodeGenerator, PlayScalaServerCodeGenerator }
+
+
+import play.grpc.gen.scaladsl.{ PlayScalaClientCodeGenerator, PlayScalaServerCodeGenerator }
+
+import play.grpc.gen.javadsl.PlayJavaServerCodeGenerator
```
Comment (ignasi35) [inline grpc-example/grpc-example-scala/build.sbt]: this seems unused and, considering it's `xxx.javadsl.xxx`, I'd say it's missplaced?

File: shopping-cart/shopping-cart-java/pom.xml [hunk 2]
```diff
@@ -221,8 +221,8 @@
 
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <lagom.version>1.6.0-RC2</lagom.version>
-        <akka.version>2.6.0-RC2</akka.version>
+        <lagom.version>1.6.0-RC3</lagom.version>
+        <akka.version>2.6.1</akka.version>
```
Comment (ignasi35) [inline shopping-cart/shopping-cart-java/pom.xml]: IIUC, Lagom `1.6.0-RC3` already uses Akka `2.6.1`. Do we need the explicit version of Akka?

Hmm, this property may be a leftover from the times when `akka-discovery` was not [pulled transitively](https://github.com/lagom/lagom-samples/blob/1.5.x/shopping-cart/shopping-cart-java/pom.xml#L192). I'd expect [any akka artifact](https://github.com/lagom/lagom-samples/blob/1259224e7e2cfb4d435584232456e6ff10610a34/shopping-cart/shopping-cart-java/pom.xml#L189-L198) in this sample to be already listed on the Lagom's BOM so there'd be no need to add them explicitly. 

I guess we can do a dependency cleanup on a separate PR.

File: grpc-example/grpc-example-scala/build.sbt [hunk 3]
```diff
@@ -62,13 +72,16 @@ lazy val `hello-proxy-api` = (project in file("hello-proxy-api"))
...
+  .enablePlugins(PlayAkkaHttp2Support) // enables serving HTTP/2 and gRPC
   .settings(
   akkaGrpcGeneratedLanguages := Seq(AkkaGrpc.Scala),
     akkaGrpcExtraGenerators += PlayScalaClientCodeGenerator,
   ).settings(
     libraryDependencies ++= Seq(
       lagomScaladslTestKit,
       macwire,
+      playGRPC,
+      akkaHttp2Support,
```
Comment (ignasi35) [inline grpc-example/grpc-example-scala/build.sbt]: Why are `PlayAkkaHttp2Support` and `akkaHttp2Support` required on the `hello-proxy-impl` project? I thought that project only built a gRPC client (in which case I find surprising that we need to enable any HTTP/2 feature).

Hmm, Ok, maybe `akkaHttp2Support` is required to build the gRPC client, but requiring `PlayAkkaHttp2Support` to consume gRPC doesn't look great. We should consider fixing this (as in "don't require `PlayAkkaHttp2Support` to only consume gRPC") in Play itself.

**Review by ignasi35 — COMMENTED** [review]
Left a few questions with doubts.

**Comment by octonato** [comment]
@ignasi35, I will clean it up. 

The reason of the imports and the `akkaHttp2Support` as lib is that I was testing the gRPC sample as well. The gRPC is out of date and I want to update it on a new PR. I failed to clean up all my local changes and end up pushing them. 

About `akkaHttp2Support`, that's was a strange one. I wasn't able to evict Akka 2.6.0 deps in the proxy and was getting an error.

**Comment by octonato** [comment]
@ignasi35, did the clean up. I already removed the direct deps on akka-discovery. That was indeed not needed.

**Comment by ignasi35** [comment]
Uh oh. Either we have an isue on the sample or there's an issue in the Lagom BOM:

```
Tests in error: 

  com.example.shoppingcart.impl.ShoppingCartReportTest: Detected possible incompatible versions on the classpath. Please note that a given Akka HTTP version MUST be the same across all modules of Akka HTTP that you are using, e.g. if you use [10.1.11] all other modules that are released together MUST be of the same version. Make sure you're using a compatible set of libraries. Possibly conflicting versions [10.1.11, 10.1.10] in libraries [akka-http-spray-json:10.1.11, akka-parsing:10.1.10, akka-http:10.1.11, akka-http-core:10.1.10]

  com.example.shoppingcart.impl.ShoppingCartServiceTest: Detected possible incompatible versions on the classpath. Please note that a given Akka HTTP version MUST be the same across all modules of Akka HTTP that you are using, e.g. if you use [10.1.11] all other modules that are released together MUST be of the same version. Make sure you're using a compatible set of libraries. Possibly conflicting versions [10.1.11, 10.1.10] in libraries [akka-http-spray-json:10.1.11, akka-parsing:10.1.10, akka-http:10.1.11, akka-http-core:10.1.10]

  com.example.shoppingcart.impl.ShoppingCartServiceTest

```

Shopping-cart (maven) fails to run tests due to mixed Akka-HTTP

**Comment by ignasi35** [comment]
Also, I think the gRPC sample is broken because of https://github.com/akka/akka-grpc/issues/731 and we'll have to wait until all upstream issues are sorted out.

**Comment by octonato** [comment]
> Uh oh. Either we have an issue on the sample or there's an issue in the Lagom BOM

Is the sample, the pom is adding akka-http 10.1.10

**Review by ignasi35 — APPROVED** [review]

File: couchbase-persistence/couchbase-persistence-scala-sbt/build.sbt [hunk 4]
```diff
@@ -14,7 +14,7 @@ val scalaTest = "org.scalatest" %% "scalatest" % "3.0.8" % Test
 lazy val `couchbase-persistence-scala-sbt` = (project in file("."))
   .aggregate(`hello-api`, `hello-impl`)
 
-val akkaVersion = "2.6.0-RC2"
+val akkaVersion = "2.6.1"
```
Comment (marcospereira) [inline couchbase-persistence/couchbase-persistence-scala-sbt/build.sbt:17]: Remove this val and use `com.lightbend.lagom.core.LagomVersion.akka` instead.

File: shopping-cart/shopping-cart-scala/project/build.properties [hunk 5]
```diff
@@ -1 +1 @@
-sbt.version=1.3.3
+sbt.version=1.2.8
```
Comment (marcospereira) [inline shopping-cart/shopping-cart-scala/project/build.properties:1]: Why downgrading here?

**Review by marcospereira — COMMENTED** [review]
Some comments after the merge.

File: couchbase-persistence/couchbase-persistence-scala-sbt/build.sbt (hunk 4 above)
Comment (ignasi35) [reply couchbase-persistence/couchbase-persistence-scala-sbt/build.sbt:17]:
```suggestion
val akkaVersion = com.lightbend.lagom.core.LagomVersion.akka
```

Comment (ignasi35) [reply couchbase-persistence/couchbase-persistence-scala-sbt/build.sbt:17]: I'm just trying out if it's possible to make a cod esuggestion on a Pr that's been merged already.

**Review by ignasi35 — COMMENTED** [review]
Feel free to ignore this suggestion. It probably won't work anyway.

**Pull request merged**